src/main/java/org/opensearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -310,10 +312,18 @@ public TypeParser(Supplier<ModelDao> modelDaoSupplier) {
     public static class KNNVectorFieldType extends MappedFieldType {
 
         int dimension;
+        String modelId;
 
         public KNNVectorFieldType(String name, Map<String, String> meta, int dimension) {
             super(name, false, false, true, TextSearchInfo.NONE, meta);
             this.dimension = dimension;
+            this.modelId = null; // By default, this is null. It will only be set for a mapping with a modelId set.
```

Comment:
This constructor should call other constructor with modelId as null

Reply:
Good point, will update